tests/components/todoist/test_calendar.py

```diff
@@ -1,70 +1,67 @@
 """Unit tests for the Todoist calendar platform."""
 from datetime import datetime
-from typing import Any
-from unittest.mock import Mock, patch
+from unittest.mock import AsyncMock, patch
 
 import pytest
+from todoist_api_python.models import Due, Label, Project
 
 from homeassistant import setup
-from homeassistant.components.todoist.calendar import DOMAIN, _parse_due_date
-from homeassistant.components.todoist.types import DueDate
+from homeassistant.components.todoist.calendar import (
+    DOMAIN,
+    _parse_due_date,
+    get_system_utc_offset_hours,
+)
 from homeassistant.const import CONF_TOKEN
 from homeassistant.helpers import entity_registry
 from homeassistant.util import dt
 
 
+@pytest.fixture(name="api")
+def mock_api() -> AsyncMock:
+    """Mock the api state."""
+    api = AsyncMock()
+    api.get_projects.return_value = [
+        Project(
+            id=12345,
+            color="blue",
+            comment_count=0,
+            favorite=False,
+            name="Name",
+            shared=False,
+            sync_id=123,
+            url="",
+        )
+    ]
+    api.get_labels.return_value = [
+        Label(id=1, name="label1", color=1, order=1, favorite=False)
+    ]
+    api.get_collaborators.return_value = []
+    return api
+
+
 def test_parse_due_date_invalid():
     """Test None is returned if the due date can't be parsed."""
-    data: DueDate = {
-        "date": "invalid",
-        "is_recurring": False,
-        "lang": "en",
-        "string": "",
-        "timezone": None,
-    }
+    data = Due(date="invalid", recurring=False, string="")
     assert _parse_due_date(data, timezone_offset=-8) is None
 
 
 def test_parse_due_date_with_no_time_data():
     """Test due date is parsed correctly when it has no time data."""
-    data: DueDate = {
-        "date": "2022-02-02",
-        "is_recurring": False,
-        "lang": "en",
-        "string": "Feb 2 2:00 PM",
-        "timezone": None,
-    }
+    data = Due(date="2022-02-02", recurring=False, string="Feb 2 2:00 PM")
     actual = _parse_due_date(data, timezone_offset=-8)
     assert datetime(2022, 2, 2, 8, 0, 0, tzinfo=dt.UTC) == actual
 
 
 def test_parse_due_date_without_timezone_uses_offset():
     """Test due date uses user local timezone offset when it has no timezone."""
-    data: DueDate = {
-        "date": "2022-02-02T14:00:00",
-        "is_recurring": False,
-        "lang": "en",
-        "string": "Feb 2 2:00 PM",
-        "timezone": None,
-    }
+    data = Due(date="2022-02-02T14:00:00", recurring=False, string="Feb 2 2:00 PM")
     actual = _parse_due_date(data, timezone_offset=-8)
     assert datetime(2022, 2, 2, 22, 0, 0, tzinfo=dt.UTC) == actual
 
 
-@pytest.fixture(name="state")
-def mock_state() -> dict[str, Any]:
-    """Mock the api state."""
-    return {
-        "collaborators": [],
-        "labels": [{"name": "label1", "id": 1}],
-        "projects": [{"id": 12345, "name": "Name"}],
-    }
```

Comment:
This fixture was replaced with [mock_api](https://github.com/home-assistant/core/pull/79481/files#diff-b97322e720d3864f9b6b0ed3f95d28858bd3b654b3c5d7c58b68779084862386R19)